doc/Tests_XML_Definition_and_Sample.md

```diff
@@ -33,6 +44,186 @@ Example:
+        </description>
+        <specs>
+          <spec>
+            <spec_assertion category="SIMD" element_name="Float32x4" element_type="method" interface="SIMD" section="SIMD" specification="SIMD"/>
+            <spec_url>https://github.com/johnmccutchan/ecmascript_simd/blob/master/README.md</spec_url>
+            <spec_statement/>
+          </spec>
+        </specs>
+      </testcase>
+    </set>
+  </suite>
+</test_definition>
+```
+#### "nodeunit" type
+
+The test load with node js unit test runner
+
+Example:
+
+```xml
+
+<?xml version="1.0" encoding="UTF-8"?>
+<?xml-stylesheet type="text/xsl" href="./testcase.xsl"?>
+<test_definition>
+    <set name="common_module" type="nodeunit">
+      <testcase component="Crosswalk App Tools/CLI" execution_type="auto" id="Crosswalk_create_no_sdk" priority="P2" purpose="Android - Validate if project is created fail without android sdk" status="approved" type="Functional">
+        <description>
+          <test_script_entry>/opt/apptools-android-tests/apptools/create_no_sdk.py</test_script_entry>
+        </description>
+      </testcase>
+    </set>
+  </suite>
+</test_definition>
+```
+###Core Test Type 
+
+#### "script" type
+
+The test case with directly script execution entry
+
+Example:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<?xml-stylesheet type="text/xsl" href="./testcase.xsl"?>
+<test_definition>
+  <suite category="Crosswalk WRT" name="wrt-sharedmode-android-tests">
+    <set name="android" type="script">
+      <testcase component="Crosswalk WRT/Shared Mode" execution_type="auto" id="Crosswalk_ShareMode_Library_Install_test" priority="P1" purpose="Check install xwalk lib" status="approved" type="Functional">
+        <description>
+          <pre_condition>
+          </pre_condition>
+          <test_script_entry test_script_expected_result="0" timeout="180">/opt/wrt-sharedmode-android-tests/sharedmode/Crosswalk_ShareMode_Library_Install_test.sh</test_script_entry>
+        </description>
+        </testcase>
+    </set>
+  </suite>
+</test_definition>
+```
+
+#### "pyunit" type
+
+The test load with pyunit test module.
+
+Example:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<?xml-stylesheet type="text/xsl" href="./testcase.xsl"?>
+<test_definition>
+  <suite category="Crosswalk App Tools" name="apptools-android-tests">
+    <set name="CLI" type="pyunit">
+      <testcase component="Crosswalk App Tools/CLI" execution_type="auto" id="Crosswalk_create_no_sdk" priority="P2" purpose="Android - Validate if project is created fail without android sdk" status="approved" type="Functional">
+        <description>
+          <test_script_entry>/opt/apptools-android-tests/apptools/create_no_sdk.py</test_script_entry>
+        </description>
+      </testcase>
+    </set>
+  </suite>
+</test_definition>
+```
+#### "androidunit" type
+
+The test load with android unit test runner
```

Comment:
=> Tests written in [androidunit](http://developer.android.com/training/testing/unit-testing/index.html) style, which ...